File: automation_web_app/FaceTrackingApi/Friend.py
import enum

from Face import Face

class Relationship(enum.Enum):
        Friend      = 1
        Family      = 2
        Colleague   = 3
        Self        = 4
        NoRelation  = 5

class Friend:
    def __init__(self, id=0, name="", relationship="", data="", encoding="", faceId=0):
        self.id             = id
        self.name           = name
        self.face           = Face(faceId, data, encoding)
        
        self.getRelation(relationship)
    
    def toDict(self):
        friendDict = {
                    'id'            : self.id
                ,   'name'          : self.name
                ,   'relationship'  : self.relationship.name
                ,   'face'          : self.face.toDict()
            }

        return friendDict

    def getRelation(self, relationship):
        if relationship.lower() == "friend":
            self.relationship = Relationship.Friend
        elif relationship.lower() == "family":
            self.relationship = Relationship.Family
        elif relationship.lower() == "colleague":
            self.relationship = Relationship.Colleague
        elif relationship.lower() == "self":
            self.relationship = Relationship.Self
        else:
            self.relationship = Relationship.NoRelation

File: automation_web_app/automation-app/src/app/data/Friend.ts
import { Face } from './Face';

export enum Relationship
{
		Friend
	,	Family
	,	Colleague
	,	Self
	,	NoRelation

};

export class Friend
{
	public id:				number;
	public name:			string;
	public relationship:	Relationship;
	public face:			Face;
	public encoding:		Array<number>;

	constructor(id: number, name: string, relationship: string, data: string, encoding: Array<number>)
	{
		this.id				= id;
		this.name			= name;
		this.face			= new Face(data, encoding);
		
		this.determineRelationship(relationship);
	}

	private determineRelationship(relationship: string): void
	{
		if (relationship.toLowerCase() == "friend"		)	this.relationship = Relationship.Friend;
		if (relationship.toLowerCase() == "family"		)	this.relationship = Relationship.Family;	else
		if (relationship.toLowerCase() == "colleague"	)	this.relationship = Relationship.Colleague; else
		if (relationship.toLowerCase() == "self"		)	this.relationship = Relationship.Self;
		else												this.relationship = Relationship.NoRelation;
	}
}